### PR TITLE
Update pydantic mip

### PR DESCRIPTION
**Pull request by henrikstranneheim**
### Added

-

### Changed

-

### Fixed

-

### How to prepare for test

- [ ] Ssh to relevant server (depending on type of change)
- [ ] Use stage: `us`
- [ ] Paxa the environment: `paxa`
- [ ] Install on stage (example for Hasta):
    ```shell
    bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_cg -t cg -b update-pydantic-mip
    ```

### How to test

- [ ] Do ...

### Expected test outcome

- [ ] Check that ...
- [ ] Take a screenshot and attach or copy/paste the output.

## Review

- [ ] Tests executed by
- [ ] "Merge and deploy" approved by
Thanks for filling in who performed the code review and the test!

### This [version](https://semver.org/) is a

- [ ] **MAJOR** - when you make incompatible API changes
- [ ] **MINOR** - when you add functionality in a backwards compatible manner
- [x] **PATCH** - when you make backwards compatible bug fixes or documentation/instructions

## Implementation Plan

- [ ] Document in ...
- [ ] Deploy this branch on ...
- [ ] Inform to ...

### CODE REVIEW

File: cg/models/mip/mip_config.py [hunk 1]
```diff
@@ -13,29 +15,33 @@ class AnalysisType(BaseModel):
...
     """This model is used when validating the mip analysis config"""
 
-    family_id_: str = Field(None, alias="family_id")
-    case_id: str = None
-    analysis_type_: dict = Field(..., alias="analysis_type")
-    samples: list[AnalysisType] = None
-    config_path: str = Field(..., alias="config_file_analysis")
-    deliverables_file_path: str = Field(..., alias="store_file")
+    family_id_: str | None = Field(None, alias="family_id")
+    case_id: Annotated[str | None, Field(validate_default=True)] = None
```
Comment (ChrOertlin) [inline cg/models/mip/mip_config.py:19]: Probably out of scope here, but this seems odd. If family_id is only supported in older mip versions shouldn't we tie a specific mip config (i..e v1) to those versions? Should we address this in some way?

**Review by ChrOertlin — APPROVED** [review]
Looks good! nice to migrate the pipeline models!

**Comment by henrikstranneheim** [comment]
### cgworkflow mip-dna start justhusky, 👍 
```
cg workflow mip-dna start justhusky --dry-run
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Called undefined __pydantic_serializer__ on HousekeeperAPI, please wrap
Called undefined __dataclass_fields__ on HousekeeperAPI, please wrap
Case justhusky exists in Status DB
Starting full MIP analysis workflow for case justhusky
Illumina run check is not applicable - the case is either down sampled or external.
Case justhusky exists in Status DB
Case justhusky exists in Status DB
Running command /home/proj/stage/bin/miniconda3/envs/S_scout/bin/scout --config /home/proj/stage/servers/config/hasta.scilifelab.se/scout-stage.yaml export panel --bed BRAIN Cardiology CILM CH CHILDHYPOTONIA CTD DIAB ENDO EP HEARING HYDRO IBMFS IEM IF mcarta MHT MIT MOVE mtDNA NBS-M NEURODEG NMD OMIM-AUTO OPTIC PANELAPP-GREEN PEDHEP PID PIDCAD RETINA SKD SOVM STROKE AID Inherited cancer --build 3

File: cg/models/mip/mip_config.py (hunk 1 above)
Comment (henrikstranneheim) [reply cg/models/mip/mip_config.py:19]: No, it should be with a modern config.

**Comment by henrikstranneheim** [comment]
Deployed to prod
```
Log deploy... done.
cg, version 63.5.15

**Pull request merged**